crates/sui-core/src/authority/authority_per_epoch_store.rs

```diff
@@ -1478,25 +1509,16 @@ impl AuthorityPerEpochStore {
         &self,
         cert: &VerifiedExecutableTransaction,
         commit_round: Round,
-        previously_deferred_tx_digests: &HashSet<TransactionDigest>,
-        last_randomness_round: RandomnessRound,
+        generating_randomness: bool,
     ) -> Option<DeferralKey> {
-        // Defer transaction if it depends on Random object.
-        if cert
-            .shared_input_objects()
-            .any(|obj| obj.id() == SUI_RANDOMNESS_STATE_OBJECT_ID)
+        // Defer transaction if it uses randomness but we are not yet generating randomness.
+        if !generating_randomness
+            && self.randomness_state_enabled()
+            && cert
+                .shared_input_objects()
+                .any(|obj| obj.id() == SUI_RANDOMNESS_STATE_OBJECT_ID)
```

Comment:
can you add a method on transaction called `is_randomness_reader()`? You have this same block of code of consensus_handler.rs

Reply:
Done